Secure Software Supply Chain (#26)
LF-Decentralized-Trust/governance: tac/governing-documents/release-signing.md

---
layout: default
title:  Release Signing
parent: Governing Documents
grand_parent: LF Decentralized Trust TAC

> nav_order: 1

Reviewer: ```suggestion
nav_order: 10
```
Not sure the right number, but I am guessing 10 is close.

Author: Do you think there's a possibility for us to improve the structure with sub-folders? For instance, project reports may have annual report and quarterly report in them. Security may have governing documents related to software supplychain, the other process related to having a representative from the project.

Reviewer 2: +1 for trying to avoid too many/too deep folder structures in general

---
[//]: # (SPDX-License-Identifier: CC-BY-4.0)

# Securing and Verifying Releases

The LF Decentralized Trust (LFDT) emphasizes a structured approach to governance, particularly in the context of artifact signing for its projects. Artifacts, which are essential outputs from each release, include binaries, container images, checksum files, documentation, provenance metadata, and software bills of materials (SBOM).

LFDT has a holistic approach to governing the security of projects and implements the highest levels of standards. The objective is to establish a consistent and secure approach to signing artifacts, leveraging industry-standard tools and practices while allowing flexibility for individual project implementations. Software verification processes are made available to ensure the integrity and authenticity of all artifacts throughout their lifecycle. A monitoring framework is introduced to score the project, rating the release process.

> This initiative aligns with efforts to establish a secure software supply chain for LFDT projects and achieve full [SLSA (Supply Chain Levels for Software Artifacts) compliance](https://slsa.dev/spec/v1.0/levels) in the release process.

Reviewer: ```suggestion
This policy aligns with efforts to establish a secure software supply chain for LFDT projects and achieve full [SLSA (Supply Chain Levels for Software Artifacts) compliance](https://slsa.dev/spec/v1.0/levels) in the release process.
```

This governance framework prioritizes understanding the "what" of artifact signing, which includes:
- The importance of signing artifacts to ensure their integrity and authenticity.
- The necessity of meeting security standards, such as SLSA compliance.

The technical "how" of implementing will be left to the discretion of individual projects, allowing for flexibility and innovation.

## LFDT Project Compliance

> 1. The LFDT Technical Advisory Committee (TAC) standardizes the signing of artifacts using tools from the Linux Foundation’s digital signing project, [OpenSSF Sigstore](https://openssf.org/projects/sigstore/).

Reviewer: ```suggestion
1. The LFDT Technical Advisory Committee (TAC) standardizes the signing of artifacts and confirmation of signing using tools from the Linux Foundation’s Open Source Security Foundation (OpenSSF) -- [OpenSSF Sigstore](https://openssf.org/projects/sigstore/) and [OpenSSF Scorecard](https://openssf.org/projects/scorecard/), respectively.
```

> 2. Each project must document the process of verifying the artifacts using [OpenSSF Sigstore](https://openssf.org/projects/sigstore/). The project must have a file called `RELEASE.md`.

Reviewer: ```suggestion
2. Each project must document the process of signing and verifying the artifacts using [OpenSSF Sigstore](https://openssf.org/projects/sigstore/) in a file called `RELEASE.md`.
```

> 3. Each project must enable and apply for a Sigstore's scorecard. The project must capture the scorecard's result and generate a badge visible on the README.md file.

Reviewer: ```suggestion
3. Each project must enable and apply for an [OpenSSF's scorecard](https://openssf.org/projects/scorecard/). The project must capture the scorecard's result and generate a badge visible on the `README.md` file.
```

> 4. An LFDT incubation project will follow at least SLSA Level 1, while graduated projects will meet at least SLSA Level 2.

Reviewer: ```suggestion
4. An LFDT incubation project must meet at least SLSA Level 1, while graduated projects must meet at least SLSA Level 2.
```

Reviewer 3: I've gone back and forth on whether graduated projects MUST sign artifacts, I think that is the main item to discuss.

## Responsibilities
- **Project Team:**
  - Add a `RELEASE.md` file containing the build and release process of artifacts. Capture the information such as versioning followed, steps to reproduce the build, current release process, resources for verifying the signed artifacts at the least.
  - Generate the scorecard and upload as a badge on `README.md` file.

>   - Each project will be responsible for the technical integration of Sigstore's cosign and scorecard into its build process. This approach allows projects to adapt the implementation to their specific needs while adhering to the overarching governance framework.

Reviewer: ```suggestion
  - Each project will be responsible for the technical integration of Sigstore's cosign and OpenSSF's scorecard into its build process. This approach allows projects to adapt the implementation to their specific needs while adhering to the overarching governance framework.
```
